content/blogs/2024-04-10-aws-data-pipeline.md

```diff
@@ -0,0 +1,198 @@
+---
+title: "Building AWS Data Pipeline with Kestra"
+description: "Put together an AWS Data Pipeline consisting of DynamoDB, S3 and Redshift using Kestra."
+date: 2024-04-10T10:00:00
+category: Solutions
+author:
+  name: Shruti Mantri
+  image: "smantri"
+---
+
+Amazon Web Services(AWS) is a leading cloud provider with heavy industry adoption. This blog post dives into [Kestra's](https://github.com/kestra-io/kestra) integrations for AWS, with an example of a real-world data pipeline. The data pipeline consists of multiple AWS services like DynamoDB, S3 and Redshift, which are put together using Kestra's orchestration capabilities.
+
+## Kestra and AWS
+
+AWS offers a vast array of cloud services, including computing power, storage, database solutions, networking and more. This extensive portfolio of service offerings is one of the key advantages of using AWS. AWS also offers a pay-as-you-go pricing model, allowing organizations to pay only for the resources they consume, thereby reducing costs and optimizing resource utilization. With these features, AWS has become the backbone of many businesses, from startups to enterprises, providing them with scalable and reliable infrastructure to innovate and grow.
+
+Kestra is a powerful orchestration engine with a rich set of plugins. Kestra seamlessly integrates with multiple [AWS services](https://kestra.io/plugins/plugin-aws) making it easy to orchestrate AWS services based data pipelines.
+
+## Usecase
+
+In this blog post, we will develop a data pipeline that has data about [products](https://raw.githubusercontent.com/kestra-io/datasets/main/csv/products.csv) and [orders](https://raw.githubusercontent.com/kestra-io/datasets/main/csv/products.csv). We start with both the data sets being present as CSV files. Our final aim is to join these two data sets and have detailed orders, where each order record has complete product information, in the CSV format.
+
+In the actual world, dimension data like products would be present in databases like RDS or DynamoDB, while fact data like orders will be present on file systems like S3. Taking this into consideration, we will have a data preparation phase where we would load the products CSV file onto DynamoDB, and orders CSV file onto S3 using Kestra.
+
+Then, we would proceed to create the data pipeline. We would load the products data fron DynamoDB onto Redshift, and load orders data from S3 onto Redshift. We would join these two tables from Redshift, and upload the detailed orders onto S3.
```

Comment:
```suggestion
Then, we will proceed to create the data pipeline. We will load the product data from DynamoDB onto Redshift and load order data from S3 onto Redshift. We'll join these two tables from Redshift and upload the detailed orders to S3.
```

@shrutimantri we discussed this in the past, don't use "would" in a tutorial. You can replace with "will" or replace with "Next...", "Let's now look at how to ..." etc